### PR TITLE
fix: exclude projen in upgrades when version is specified

### PR DESCRIPTION
**Pull request by dontirun**
Also fixes  https://github.com/projen/projen/issues/1356#issuecomment-1005046584 

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by eladb** [comment]
@iliapolo can you take a look?

File: src/javascript/node-project.ts [hunk 1]
```diff
@@ -455,6 +456,11 @@ export class NodeProject extends GitHubProject {
    */
   public readonly upgradeWorkflow?: UpgradeDependencies;
 
+  /**
+   * The packages that are ignored during the upgrade workflow.
+   */
+  public readonly ignoredUpgrades: string[] = [];
```
Comment (iliapolo) [inline src/javascript/node-project.ts:462]: This functionality actually already exists:

https://github.com/projen/projen/blob/1e0bdedcd2e3a0366ef3143d7b7207bf1d58d4b1/src/javascript/upgrade-dependencies.ts#L23-L29

You can configure it on a node project via:

```ts
depsUpgradeOptions: {
  exclude: ['projen']
}
```

File: src/javascript/node-project.ts [hunk 2]
```diff
@@ -709,6 +715,14 @@ export class NodeProject extends GitHubProject {
       );
       ignoresProjen = upgradeDependencies.ignoresProjen;
       this.upgradeWorkflow = upgradeDependencies;
+      new YamlFile(this, ".ncurc.yml", {
+        obj: {
+          reject: this.ignoredUpgrades,
+        },
+      });
+      if (options.projenVersion) {
```
Comment (iliapolo) [inline src/javascript/node-project.ts:723]: I agree that if a user specified `projenVersion` we should add `projen` to the ignore list. But this should be done here:

https://github.com/projen/projen/blob/1e0bdedcd2e3a0366ef3143d7b7207bf1d58d4b1/src/javascript/node-project.ts#L693

**Review by iliapolo — CHANGES_REQUESTED** [review]
@dontirun Thanks for this. See my comments. 

We should repurpose this PR to simply ignore `projen` by default if `projenVersion` is defined. Also this needs a test.

File: src/javascript/node-project.ts (hunk 1 above)
Comment (dontirun) [reply src/javascript/node-project.ts:462]: I completely missed that 🤦. I'll redothe PR

File: src/javascript/node-project.ts (hunk 2 above)
Comment (dontirun) [reply src/javascript/node-project.ts:723]: If we specify it here we should make sure that it's not overwritten by the [deepMerge](
https://github.com/dontirun/projen/blob/1588d8c1ea69fd536ef1ab03e7df441850497d1f/src/javascript/node-project.ts#L714).

**Comment by dontirun** [comment]
@iliapolo Is this something that is still wanted/needed?

File: src/javascript/node-project.ts [hunk 3]
```diff
@@ -703,16 +703,31 @@ export class NodeProject extends GitHubProject {
           gitIdentity: this.workflowGitIdentity,
         },
       };
+      const depsUpgradeOptions = options.depsUpgradeOptions ?? {};
+      const projenExclude = depsUpgradeOptions?.exclude ?? [];
+      if (options.projenVersion && !projenExclude.includes("projen")) {
+        projenExclude.push("projen");
+      }
```
Comment (iliapolo) [inline src/javascript/node-project.ts:710]: This feels a bit cluttered. As I see it, all we need to do is provide a default value to `ignoreProjen`, both in the dependabot config and the dep upgrade. 

So, I think we just need to add: 

```ts
ignoreProjen: !!options.projenVersion,
```

Here:

https://github.com/projen/projen/blob/fafd5056e572185359e19d50b685f37ed496d1aa/src/javascript/node-project.ts#L693-L695

And here:

https://github.com/projen/projen/blob/fafd5056e572185359e19d50b685f37ed496d1aa/src/javascript/node-project.ts#L703-L705

No?

File: src/javascript/node-project.ts [hunk 4]
```diff
@@ -703,16 +703,31 @@ export class NodeProject extends GitHubProject {
...
+        ])
       );
       ignoresProjen = upgradeDependencies.ignoresProjen;
       this.upgradeWorkflow = upgradeDependencies;
     }
 
     // create a dedicated workflow to upgrade projen itself if needed
-    if (ignoresProjen && this.package.packageName !== "projen") {
+    if (
+      !options.projenVersion &&
```
Comment (iliapolo) [inline src/javascript/node-project.ts:727]: If we apply the above suggestion, we won't need to add this condition here.

**Review by iliapolo — CHANGES_REQUESTED** [review]
@dontirun Sorry for letting this hang for a while. 

See my comments.

**Pull request closed without merging**